src/main/java/com/epam/test_generator/config/security/JwtAuthenticationFilter.java

```diff
@@ -1,67 +1,116 @@
-
-
-    public JwtAuthenticationFilter() {
-        super("/**");
+import org.codehaus.jackson.map.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * Security filter that performs jwt authentication only if valid Authorization header is present.
+ * If token is not provided, request execution is continued and user would be soon marked anonymous
+ * by {@link org.springframework.security.web.authentication.AnonymousAuthenticationFilter}. If
+ * token is present, filter tries to parse it and if it is invalid, error returned immediately no
+ * matter if request was made to guarded resource or not. On success, {@link Authentication}
+ * object is created and placed to SecurityContextHolder.
+ */
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private LoginService loginService;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private ProjectService projectService;
+
+    public JwtAuthenticationFilter(){
     }
 
-
-    /**
-     * This method try to get Token as String from header or from the parameters. In case of sending
-     * token in Header you should set header name "Authorization" and concatenate your token with
-     * "Bearer " string. If token was successfully received, method will call standard
-     * Spring-Security Authenticated_Manager which will call {@link JwtAuthenticationProvider} which
-     * will return {@link AuthenticatedUser} object.
-     */
     @Override
-    public Authentication attemptAuthentication(HttpServletRequest request,
-                                                HttpServletResponse response)
-        throws AuthenticationException, IOException, ServletException {
-
-        String token;
-        String authorization = request.getHeader("Authorization");
-        if (authorization != null) {
-            token = authorization.replaceFirst("^Bearer ", "");
-        } else {
-            token = request.getParameter("token");
+    protected void doFilterInternal(
+        HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+        throws ServletException, IOException {
+
+        String header = request.getHeader("Authorization");
+        if (header == null) {
+            chain.doFilter(request, response);
+            return;
         }
-        if (token == null) {
-            throw new TokenMissingException("No JWT token found in request headers");
+        try {
+            String token = header.replaceAll("^Bearer ", "");
+
+            DecodedJWT decodedJwt = loginService.validate(token);
```

Comment:
Method name and returned value name are a bit confusing. You calling `loginService.validate(token)`, which should do some validation according to the method name. But instead it does decoding of that token.  I'd name that method `decode` instead of `validate`.